main.tf

```
@@ -2,11 +2,11 @@ locals {
   enabled = module.this.enabled
 
   # See https://aws.amazon.com/blogs/containers/introducing-launch-template-and-custom-ami-support-in-amazon-eks-managed-node-groups/
-  features_require_ami    = local.enabled && local.need_bootstrap
-  configured_ami_image_id = var.ami_image_id == null ? "" : var.ami_image_id
-  need_ami_id             = local.enabled ? local.features_require_ami && length(local.configured_ami_image_id) == 0 : false
-
-  features_require_launch_template = local.enabled ? length(var.resources_to_tag) > 0 || local.need_userdata || local.features_require_ami : false
+  features_require_ami               = local.enabled && local.need_bootstrap
+  configured_ami_image_id            = var.ami_image_id == null ? "" : var.ami_image_id
+  need_ami_id                        = local.enabled ? local.features_require_ami && length(local.configured_ami_image_id) == 0 : false
+  features_require_imds_restrictions = var.launch_template_metadata_options_http_endpoint == "enabled" && var.launch_template_metadata_options_http_put_response_hop_limit == 1 && var.launch_template_metadata_options_http_tokens == "required"
+  features_require_launch_template   = local.enabled ? length(var.resources_to_tag) > 0 || local.need_userdata || local.features_require_ami || local.features_require_imds_restrictions : false
```

This is wrong. We need to use a launch template to set non-default values of IMDS. So something like this:

```suggestion
  need_imds_settings = var.launch_template_metadata_options_http_endpoint != "enabled" || var.launch_template_metadata_options_http_put_response_hop_limit != 1 || var.launch_template_metadata_options_http_tokens != "optional"
  features_require_launch_template   = local.enabled ? length(var.resources_to_tag) > 0 || local.need_userdata || local.features_require_ami || local.need_imds_settings : false
```